mongodb/bucketarchives.go

```diff
@@ -19,13 +19,32 @@ type Archives struct {
 }
 
 type Archive struct {
-	Cid        []byte `bson:"cid"`
-	JobID      string `bson:"job_id"`
-	JobStatus  int    `bson:"job_status"`
-	Aborted    bool   `bson:"aborted"`
-	AbortedMsg string `bson:"aborted_msg"`
-	FailureMsg string `bson:"failure_msg"`
-	CreatedAt  int64  `bson:"created_at"`
+	Cid        []byte     `bson:"cid"`
+	JobID      string     `bson:"job_id"`
+	Status     int        `bson:"status"`
+	Aborted    bool       `bson:"aborted"`
+	AbortedMsg string     `bson:"aborted_msg"`
+	FailureMsg string     `bson:"failure_msg"`
+	CreatedAt  int64      `bson:"created_at"`
+	DealInfo   []DealInfo `bson:"deal_info"`
```

Comment:
New model to hold deal information. This would just unmarshal to an empty array for pre-existing archives I think? Will look at writing a test for that. Also, maybe we want to consider writing a migration for those pre-existing archives. Not sure how complex that could be.

Reply:
Yeah, I'd think the same: unmarshal to empty array for existing things.

Regarding the migration, could be. We could wait a bit to see if existing archives really start asking for this data to be populated, and if justified do it?

Reply:
Sounds like a plan to me.